File: src/lib/queries/comments.ts
import { useMutation, useQuery, useQueryClient } from "@tanstack/react-query";
import { api } from "@/lib/tauri";
import type { NewComment } from "@/types/generated";

export const commentKeys = {
  byTask: (taskId: string) => ["comments", "task", taskId] as const,
};

export function useComments(taskId: string) {
  return useQuery({
    queryKey: commentKeys.byTask(taskId),
    queryFn: () => api.listComments(taskId),
    enabled: !!taskId,
  });
}

export function useCreateComment(taskId: string) {
  const qc = useQueryClient();
  return useMutation({
    mutationFn: (input: NewComment) => api.createComment(input),
    onSuccess: () =>
      qc.invalidateQueries({ queryKey: commentKeys.byTask(taskId) }),
  });
}

export function useDeleteComment(taskId: string) {
  const qc = useQueryClient();
  return useMutation({
    mutationFn: (id: string) => api.deleteComment(id),
    onSuccess: () =>
      qc.invalidateQueries({ queryKey: commentKeys.byTask(taskId) }),
  });
}

File: src-tauri/src/commands/comments.rs
use tauri::State;

use crate::db;
use crate::domain::{Comment, NewComment};
use crate::error::AppResult;
use crate::state::AppState;

#[tauri::command]
pub async fn list_comments(
    state: State<'_, AppState>,
    task_id: String,
) -> AppResult<Vec<Comment>> {
    db::comments::list_for_task(&state.db, &task_id).await
}

#[tauri::command]
pub async fn create_comment(
    state: State<'_, AppState>,
    input: NewComment,
) -> AppResult<Comment> {
    db::comments::create(&state.db, input).await
}

#[tauri::command]
pub async fn delete_comment(state: State<'_, AppState>, id: String) -> AppResult<()> {
    db::comments::soft_delete(&state.db, &id).await
}
